spec/services/purchase_order_creator_spec.rb

```diff
@@ -0,0 +1,39 @@
+require 'rails_helper'
+
+RSpec.describe 'PurchaseOrderCreator' do
+  describe '#call' do
+    let(:cart) { create(:cart) }
+    let(:user) { cart.user }
+    let(:address) { nil }
```

Comment:
acho que esse também não precisa, pode apenas chamar nil no único lugar que você referencia `address`

Reply:
pode também criar uma factory aqui https://github.com/feperessim/refactoring-exercise-for-trainees/tree/master/spec/factories e testar melhor o repasse desses valores.